octomap/octomap.pyx

```diff
@@ -686,6 +688,43 @@ cdef class OcTree:
         """
         self.thisptr.toMaxLikelihood()
 
+    def updateNodes(self, values, update, lazy_eval=False):
+        """
+        Integrate occupancy measurements and Manipulate log_odds value of voxel directly. 
+        """
+        if values is None or len(values) == 0:
+            return
+        if isinstance(values[0], OcTreeKey):
+            if isinstance(update, bool):
+                for v in values:
+                    self.thisptr.updateNode(defs.OcTreeKey(<unsigned short int>v[0],
+                                                           <unsigned short int>v[1],
+                                                           <unsigned short int>v[2]),
+                                            <cppbool>update,
+                                            <cppbool?>lazy_eval)
+            else:
+                for v in values:
+                    self.thisptr.updateNode(defs.OcTreeKey(<unsigned short int>v[0],
+                                                           <unsigned short int>v[1],
+                                                           <unsigned short int>v[2]),
+                                            <float?>update,
+                                            <cppbool?>lazy_eval)
+        else:
+            if isinstance(update, bool):
+                for v in values:
+                    self.thisptr.updateNode(<double?>v[0],
+                                            <double?>v[1],
+                                            <double?>v[2],
+                                            <cppbool>update,
+                                            <cppbool?>lazy_eval)
+            else:
+                for v in values:
+                    self.thisptr.updateNode(<double?>v[0],
+                                            <double?>v[1],
+                                            <double?>v[2],
+                                            <float?>update,
+                                            <cppbool?>lazy_eval)
+
```

Comment:
@neka-nat maybe not needed much but i added so only one `py` -> `c` call for adding many nodes @ once (~ batch update nodes). it seemed a little faster (e.g. 2-3% for maps i was building) but i can remove if you prefer not to do it this way.